### ISSUE TITLE
EarlyStopping should have a restore_best_weights arg as in Keras

### ISSUE THREAD

**Issue by cedricrommel** [skorch-dev/skorch#808]
The EarlyStopping callback is very useful to avoid overfitting as it allows to stop training before the validation loss (or any other metric of interest) starts degrading. The problem I see with Skorch's implementation is that such callback only stops the training after the desired patience is reached, keeping the model at its state from the last epoch. It does not allow the user to reload the best model from `patience` epochs before.

This can be problematic for models where the training is noisy (often the case), which need large patience values to be properly regularized. In these cases, the model obtained at the end might have overfitted quite a lot already. This is why other user-friendly deep learning libraries such as Keras, which also implement the EarlyStopping callback, have an argument `restore_best_weights` which allows to restore the best model, from `patience` epochs before earlystopping. I really believe that such an option should be added to the `EarlyStopping` implementation in Skorch, not only because it is useful, but also because it would be coherent with Keras, which is very popular.

**Comment by BenjaminBossan** [comment]
The PR was merged, thus this issue can be closed.
